hypertrace-trace-enricher/enriched-span-constants/src/main/java/org/hypertrace/traceenricher/enrichedspan/constants/utils/EnrichedSpanUtils.java

```diff
@@ -367,11 +392,111 @@ public static List<String> getSpaceIds(Event event) {
+  }
+
+  /**
+   * The response headers are populated as key-value pair
+   *
+   * @return map of headers with identifying prefix removed key -> value
+   *     <ul>
+   *       <li>header1 -> value1
+   *       <li>header2 -> value2
+   *       <li>header3 -> value3
+   *     </ul>
+   */
+  public static Map<String, String> getResponseHeadersExceptCookies(Event event) {
+    Map<String, String> responseHeadersExceptCookies = new HashMap<>();
+
+    // To get http response headers except cookies we use http.response.header. as prefix
+    responseHeadersExceptCookies.putAll(
+        HttpSemanticConventionUtils.getHttpHeadersExceptCookies(
+            event,
+            entry -> !EnrichedSpanUtils.isHttpResponseCookie(entry.getKey()),
+            HTTP_RESPONSE_HEADER_PREFIX));
+
+    // To get grpc response headers except cookies we use grpc.response.metadata. as prefix
+    responseHeadersExceptCookies.putAll(
+        RpcSemanticConventionUtils.getGrpcHeadersExceptCookies(
+            event, GRPC_RESPONSE_METADATA_PREFIX));
+
+    // To get rpc response headers except cookies we use rpc.response.metadata. as prefix
+    responseHeadersExceptCookies.putAll(
+        RpcSemanticConventionUtils.getRpcHeadersExceptCookies(event, RPC_RESPONSE_METADATA_PREFIX));
+    return Collections.unmodifiableMap(responseHeadersExceptCookies);
+  }
+
+  /**
+   * The request cookies are populated as `http.request.header.cookie` with value as
+   * cookie1=value1;cookie2=value2;cookie3=value3
+   *
+   * @return map of cookie key -> value
+   *     <ul>
+   *       <li>cookie1 -> value1
+   *       <li>cookie2 -> value2
+   *       <li>cookie3 -> value3
+   *     </ul>
+   */
+  public static Map<String, String> getHttpRequestCookies(Event event) {
+    return HttpSemanticConventionUtils.getHttpRequestCookies(event);
+  }
+
+  /**
+   * The response cookies are populated as `http.response.header.set-cookie[0] -> cookie1=value1`,
+   * `http.response.header.set-cookie[1] -> cookie2=value2` and so on in span attributes
+   *
+   * @return map of cookie key -> value *
+   *     <ul>
+   *       *
+   *       <li>cookie1 -> value1
+   *       <li>cookie2 -> value2
+   *     </ul>
+   */
+  public static Map<String, String> getHttpResponseCookies(Event event) {
+    return HttpSemanticConventionUtils.getHttpResponseCookies(event);
+  }
+
   /** Check whether these spans belongs to different services. */
   public static boolean areBothSpansFromDifferentService(Event event, Event parentEvent) {
     if (event.getServiceName() == null || parentEvent.getServiceName() == null) {
       return false;
     }
     return !StringUtils.equals(event.getServiceName(), parentEvent.getServiceName());
   }
+
+  public static boolean isHttpRequestCookie(String requestHeaderAttributeKey) {
+    return requestHeaderAttributeKey.equals(REQUEST_COOKIE_HEADER_KEY);
+  }
+
+  public static boolean isHttpResponseCookie(String responseHeaderAttributeKey) {
```

Comment:
We already have `isHttpResponseCookie` defined in `HttpSemanticConventionUtils`. Can we please move both `isHttpRequestCookie` and `isHttpResponseCookie` to `HttpSemanticConventionUtils`?